src/java/org/apache/cassandra/metrics/DecayingEstimatedHistogramReservoir.java

```diff
@@ -250,7 +248,7 @@ public void update(long value)
 
         int index = findIndex(bucketOffsets, value);
 
-        updateBucket(decayingBuckets, index, Math.round(forwardDecayWeight(now)));
+        decayingBuckets.update(index, now);
```

Comment:
I'm on the fence with this one.
On the one hand, I don't want to do a direct `decayingBuckets.buckets` update because the weight may be given incorrectly.
On the other, it looks inconsistent with updating regular buckets, and there is no separation of concerns in DEHR anyway.

Reply:
Well, personally I don't like the lack of symmetry here as well, but, well, decayingBuckets does more than regular buckets and it had to be extracted in order to be able to set the reference atomically. Extracting this to a separate method makes the code more readable and less error prone.